src/adc.rs

```diff
@@ -456,6 +458,59 @@ impl_adc!(ADC1: adc_unit_t_ADC_UNIT_1);
 #[cfg(not(any(esp32c2, esp32h2, esp32c5, esp32c6, esp32p4)))] // TODO: Check for esp32c5 and esp32p4
 impl_adc!(ADC2: adc_unit_t_ADC_UNIT_2);
 
+/// One-shot ADC module
+/// Example: reading a value form a pin and printing it on the terminal
+/// ```
+/// use std::thread;
+/// use std::time::Duration;
+
+/// #[cfg(not(any(feature = "adc-oneshot-legacy", esp_idf_version_major = "4")))]
+/// fn main() -> anyhow::Result<()> {
+///     use esp_idf_hal::adc::attenuation::DB_11;
+///     use esp_idf_hal::adc::oneshot::config::AdcChannelConfig;
+///     use esp_idf_hal::adc::oneshot::*;
+///     use esp_idf_hal::peripherals::Peripherals;
+
+///     let peripherals = Peripherals::take()?;
+
+///     #[cfg(not(esp32))]
+///     let adc = AdcDriver::new(peripherals.adc1)?;
+
+///     #[cfg(esp32)]
+///     let adc = AdcDriver::new(peripherals.adc2)?;
+
+///     /// configuring pin to analog read, you can regulate the adc input voltage range depending on your need
+///     /// for this example we use the attenuation of 11db which sets the input voltage range to around 0-3.6V
+///     let config = AdcChannelConfig {
+///         attenuation: DB_11,
+///         calibration: true,
+///         ..Default::default()
+///     };
+
+///     #[cfg(not(esp32))]
+///     let mut adc_pin = AdcChannelDriver::new(&adc, peripherals.pins.gpio2, &config)?;
+
+///     #[cfg(esp32)]
+///     let mut adc_pin = AdcChannelDriver::new(&adc, peripherals.pins.gpio12, &config)?;
+
+///     loop {
+///         /// you can change the sleep duration depending on how often you want to sample
+///         thread::sleep(Duration::from_millis(100));
+///         println!("ADC value: {}", adc.read(&mut adc_pin)?);
+///     }
+/// }
+
+/// #[cfg(any(feature = "adc-oneshot-legacy", esp_idf_version_major = "4"))]
```

Comment:
This part does not need to be included-

Reply:
Thank you so much for the quick look. The section should be removed now.